### PR TITLE
Revert revert dvkvs invalidator

### PR DESCRIPTION
**Pull request by sudiksha27**
**Goals (and why)**:
Revert revert #5010 and remove the incorrect hasColumn check.

**Implementation Description (bullets)**:

- Remove hasColumn check

- Remove fail fast if db is poisoned

**Testing (What was existing testing like?  What have you done to improve it?)**:
Tests in 5010 should suffice

**Concerns (what feedback would you like?)**:

- Oracle tests

**Where should we start reviewing?**:

- InvalidationRunner.java

**Priority (whenever / two weeks / yesterday)**:
Monday would be ideal

### CODE REVIEW

**Review by sudiksha27 — COMMENTED** [review]
Need to cut RC for testing on internal product.

**Comment by sudiksha27** [comment]
Tests run on internal Oracle setup to verify invalidator works for Oracle backed stacks.

File: atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/InvalidationRunner.java [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+
+    private Long poisonStoreAndGetLastAllocatedTimestamp(Connection connection, Limits limits,
+            TableStatus tableStatus) throws SQLException {
+        long lastAllocated;
+
+        if (tableStatus == TableStatus.NO_DATA) {
+            lastAllocated = AtlasDbFactory.NO_OP_FAST_FORWARD_TIMESTAMP;
+        } else {
+            lastAllocated = limits.upperLimit().get().value();
+        }
```
Comment (jeremyk-91) [inline atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/InvalidationRunner.java:107]: nit:
```suggestion
        long lastAllocated = tableStatus == TableStatus.NO_DATA
                ? AtlasDbFactory.NO_OP_FAST_FORWARD_TIMESTAMP
                : limits.upperLimit().get().value();
```

File: atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/InvalidationRunner.java [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+
+        poisonTable(connection);
+        return lastAllocated;
+    }
+
+    private void poisonTable(Connection connection) throws SQLException {
+        try (Statement statement = connection.createStatement()) {
+            if (ConnectionDbTypes.getDbType(connection).equals(DBType.ORACLE)) {
```
Comment (jeremyk-91) [inline atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/InvalidationRunner.java:115]: nit: enums are usually compared with `==` as opposed to `Objects::equals`

File: atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/InvalidationRunner.java [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+    }
+
+    private TableStatus getTableStatus(Limits limits) {
+        boolean upperLimitExists = limits.upperLimit().isPresent();
+        boolean legacyUpperLimitExists = limits.legacyUpperLimit().isPresent();
+
+        if (upperLimitExists) {
+            return legacyUpperLimitExists ? TableStatus.BOTH_COLUMNS : TableStatus.HEALTHY;
+        }
+        return legacyUpperLimitExists ? TableStatus.POISONED : TableStatus.NO_DATA; // no data in table
```
Comment (jeremyk-91) [inline atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/InvalidationRunner.java:191]: nit: I don't think the comment is needed? Seems self-explanatory

File: atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/timestamp/PhysicalBoundStoreDatabaseUtils.java [hunk 4]
```diff
@@ -63,17 +63,27 @@ private static void createTimestampTableIgnoringAlreadyExistsError(
         }
     }
 
+    public static boolean tableAlreadyExistsError(SQLException exception) {
```
Comment (jeremyk-91) [inline atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/timestamp/PhysicalBoundStoreDatabaseUtils.java:66]: nit: These boolean methods usually have verbs indicating they are predicates e.g. `isTableAlreadyExistsError(...)`

**Review by jeremyk-91 — APPROVED** [review]
Broadly seems correct, only minor suggestions. Looks good

**Pull request merged**